rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_publisher.hpp

```diff
@@ -0,0 +1,171 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__LIFECYCLE_PUBLISHER_HPP_
+#define RCLCPP__LIFECYCLE_PUBLISHER_HPP_
+
+#include "rclcpp/publisher.hpp"
+
+namespace rclcpp
+{
+namespace publisher
+{
+
+namespace lifecycle_interface
+{
+/**
+ * @brief base class with only
+ * pure virtual functions. A managed
+ * node can then deactivate or activate
+ * the publishing.
+ * It is more a convenient interface class
+ * than a necessary base class.
+ */
+class PublisherInterface
```

Comment:
"ManagedLifecyclePublisherInterface" rather than using a namespace? Although as a pattern begins to emerge the use of a namespace doesn't look as bad, I don't think it is necessary nor as neat as naming the interface classes directly.